package/com.unity.formats.usd/Tests/Editor/USDPayloadComponentTests.cs

```diff
@@ -0,0 +1,207 @@
+﻿using UnityEngine;
+using UnityEngine.TestTools;
+using USD.NET;
+using System.IO;
+using UnityEditor;
+using NUnit.Framework;
+using System.Collections;
+
+namespace Unity.Formats.USD.Tests
+{
+    public class USDPayloadComponentTests
+    {
+        const string k_USDGUID = "5f0268198d3d7484cb1877bec2c5d31f"; // GUID of test_collections.usda
+ 
+        private GameObject m_usdRoot;
+        private UsdAsset m_usdAsset;
+
+        [SetUp]
+        public void Setup()
+        {
+            InitUsd.Initialize();
+            var usdPath = Path.GetFullPath(AssetDatabase.GUIDToAssetPath(k_USDGUID));
+            var stage = pxr.UsdStage.Open(usdPath, pxr.UsdStage.InitialLoadSet.LoadNone);
+            var scene = Scene.Open(stage);
+            m_usdRoot = USD.UsdMenu.ImportSceneAsGameObject(scene);
+            scene.Close();
+
+            m_usdAsset = m_usdRoot.GetComponent<UsdAsset>();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            m_usdAsset.DestroyAllImportedObjects();
+            GameObject.DestroyImmediate(m_usdRoot);
+        }
+
+        [UnityTest]
+        public IEnumerator IsLoaded_DefaultIsFalse_WhenPolicyIsSetToDontLoad()
+        {
+            // Prepare states
+            Assume.That(m_usdAsset, Is.Not.Null, "Could not find USDAsset component on root gameobject.");
+
+            m_usdAsset.m_payloadPolicy = PayloadPolicy.DontLoadPayloads;
+            m_usdAsset.Reload(forceRebuild: true);
+
+            yield return null;
+
+            // Tests
+            var usdPayload = m_usdRoot.GetComponentInChildren<UsdPayload>();
+            Assume.That(usdPayload, Is.Not.Null, "Could not find USDPayload in the subtree.");
+            Assert.That(usdPayload.IsLoaded, Is.False, "UsdPayload.IsLoaded should be set to false.");
+        }
+
+        [UnityTest]
+        public IEnumerator IsLoaded_DefaultIsTrue_WhenPolicyIsSetToLoadAll()
+        {
+            // Prepare states
+            Assume.That(m_usdAsset, Is.Not.Null, "Could not find USDAsset component on root gameobject.");
```

Comment:
I think you could add this to setup, then you won't need to check at the beginning of each test